source/common/upstream/upstream_impl.cc

```diff
@@ -1004,13 +1004,22 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(
 
   // Remove hosts from current_priority_hosts that were matched to an existing host in the previous
   // loop.
-  current_priority_hosts.erase(std::remove_if(current_priority_hosts.begin(),
-                                              current_priority_hosts.end(),
-                                              [&existing_hosts_for_current_priority](auto host) {
-                                                return existing_hosts_for_current_priority.count(
-                                                    host->address()->asString());
-                                              }),
-                               current_priority_hosts.end());
+  for (auto itr = current_priority_hosts.begin(); itr != current_priority_hosts.end();) {
+    auto existing_itr = existing_hosts_for_current_priority.find((*itr)->address()->asString());
+
+    if (existing_itr != existing_hosts_for_current_priority.end()) {
+      existing_hosts_for_current_priority.erase(existing_itr);
+      current_priority_hosts.erase(itr);
```

Comment:
In this branch, how does `itr` advance after the erase?

Reply:
After reading the docs more thoroughly I'm surprised this even works, sounds like `itr` should be invalidated when I call erase. Might be that it happens to work because the iterator is pointing to the same index as before the deletion, which becomes the index of the next element after it gets shuffled down 

Seems like what I should be doing is setting `itr` to the return value of `erase`